### PR TITLE
[41]:... unit tests for sender and receiver errors

### PR DESCRIPTION
**Pull request by Darrellbor**
PR for #41 
Implemented 
- [x] `TestReceiver_Bind`
- [x] `TestReceiver_ReadFloats`
- [x] `TestSender_Connect`
- [x] `TestSender_WriteFloats`

### CODE REVIEW

**Comment by Darrellbor** [comment]
@gavv @ortex PR ready for review

**Comment by gavv** [comment]
Also @Asalle

**Comment by ortex** [comment]
Nice work!

a couple of tests could be added:

- for Sender.Connect, Receiver.Bind: add tests with an invalid endpoint that fails to convert to `C endpoint`, e.g. with invalid protocol `Protocol(1)`
- for Sender.Writefloat, Receiver.ReadFloats: add tests with zero-sized frame `[]float32{}`

also, point for discussion (@gavv ):

Maybe add test `Receiver_Closed` (or something like that) and remove `receiverClosedBefore` scenario from all tests?
Because we need to check that each function call (except `Сlose()`) on a closed receiver return an error 
e.g:
```go

func TestReceiver_Closed(t *testing.T) {
	tests := []struct {
		name      string
		operation func(receiver *Receiver) error
	}{
		{
			name: "SetMulticastGroup after close",
			operation: func(receiver *Receiver) error {
				return receiver.SetMulticastGroup(SlotDefault, InterfaceAudioSource, "127.0.0.1")
			},
		},
		{
			name: "Bind after close",
			operation: func(receiver *Receiver) error {
				return receiver.Bind(SlotDefault, InterfaceAudioSource, nil)
			},
		},
		{
			name: "ReadFloats after close",
			operation: func(receiver *Receiver) error {
				recFloats := make([]float32, 2)
				return receiver.ReadFloats(recFloats)
			},
		},
	}
	for _, tt := range tests {
		ctx, err := OpenContext(ContextConfig{})
		require.NoError(t, err)
		require.NotNil(t, ctx)

		receiver, err := OpenReceiver(ctx, makeReceiverConfig())
		require.NoError(t, err)
		require.NotNil(t, receiver)

		err = receiver.Close()
		require.NoError(t, err)

		require.Equal(t, errors.New("receiver is closed"), tt.operation(receiver))

		err = ctx.Close()
		require.NoError(t, err)
	}
}
```

**Comment by gavv** [comment]
I agree with Andrey, separating this check will make the table tests more uniform, which is the point of tables. Ideally table tests should have minimal amount of conditions and if tests have rather different flow, they should belong to different tables / tests.

**Comment by Darrellbor** [comment]
@gavv @ortex 
reviewed based on the comments and ready for re-review

File: roc/sender_test.go [hunk 1]
```diff
@@ -103,25 +92,194 @@ func TestSender_SetReuseaddr(t *testing.T) {
...
+			err = sender.Close()
+			require.NoError(t, err)
+
 			err = ctx.Close()
 			require.NoError(t, err)
 		})
 	}
 }
+
+func TestSender_Close(t *testing.T) {
```
Comment (ortex) [inline roc/sender_test.go:243]: can we add test for `SetReuseaddr` as well?

File: roc/sender_test.go [hunk 2]
```diff
@@ -103,25 +92,194 @@ func TestSender_SetReuseaddr(t *testing.T) {
...
+			wantErr:  newNativeErr("roc_endpoint_set_protocol()", -1),
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, err := OpenContext(ContextConfig{})
+			require.NoError(t, err)
+
+			sender, err := OpenSender(ctx, SenderConfig{
```
Comment (ortex) [inline roc/sender_test.go:157]: we could use `makeSenderConfig` here as in the receiver test

**Review by ortex — APPROVED** [review]
